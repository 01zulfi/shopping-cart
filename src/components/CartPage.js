import React, { useState } from "react";
import CartItem from "./CartItem";
import cart from "../CartManager";

const CartPage = () => {
  const [items, setItems] = useState(cart.cart);
  const [total, setTotal] = useState(cart.total);

  const setTotalHandler = (value) => setTotal(value);

  const deleteButtonHandler = (event) => {
    const id = Number(event.target.getAttribute("data-id"));
    cart.remove(id);
    setTotalHandler(cart.total);
    setItems(cart.cart);
  };

  const mapCallback = (item) => {
    return (
      <div className="cart-item-wrapper" key={item.id}>
        <CartItem
          name={item.name}
          price={item.price}
          id={item.id}
          quantity={item.quantity}
          setTotalHandler={setTotalHandler}
        />
        <button
          data-id={item.id}
          className="delete-item"
          onClick={deleteButtonHandler}
        >
          Delete
        </button>
        <hr />
      </div>
    );
  };

  return (
    <div className="cart">
      <h1>Cart</h1>
      {items.map(mapCallback)}
      <h3>{total}</h3>
    </div>
  );
};

export default CartPage;
