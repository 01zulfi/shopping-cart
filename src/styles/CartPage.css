div.cart {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 95%;
}

div.cart-item-wrapper,
div.checkout {
  display: flex;
  flex-direction: column;
  align-items: center;
  border: 1px solid #d4bff9;
  margin-bottom: 10px;
  padding: 15px;
  background: rgba(255, 255, 255, 0.09);
}

div.cart-item-wrapper button,
div.checkout button {
  animation: button-animation 3s infinite linear alternate;
  padding: 10px;
  border: 1px solid #d4bff9;
  color: black;
  border: 0;
  font-size: 20px;
  border-radius: 5px;
}

@keyframes button-animation {
  from {
    background-color: #bc75c9;
  }
  to {
    background-color: #ab47bc;
  }
}

div.checkout {
  width: 95%;
}
