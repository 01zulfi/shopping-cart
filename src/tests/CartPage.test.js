import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import CartPage from "../components/CartPage";

test("cart renders", () => {
  const { container } = render(<CartPage />);
  expect(container).toMatchSnapshot();
  expect(container.firstChild).toHaveClass("cart");
});
