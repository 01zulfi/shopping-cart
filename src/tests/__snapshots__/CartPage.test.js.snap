// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cart renders 1`] = `
<div>
  <div
    class="cart"
  >
    <h1>
      Cart
    </h1>
    <h3>
      0
    </h3>
  </div>
</div>
`;
